test/e2e/webdriver/driver.js

```diff
@@ -1164,6 +1164,19 @@ class Driver {
     return await this.driver.wait(until.urlIs(url), timeout);
   }
 
+  /**
+   * Waits until the current URL includes the specified substring.
+   *
+   * @param {object} options - Parameters for the function.
+   * @param {string} options.url - Substring to wait for in the URL.
+   * @param {int} options.timeout - optional timeout period, defaults to this.timeout.
+   * @returns {Promise<void>} Promise that resolves once the URL includes the substring.
+   * @throws {Error} Throws an error if the URL does not include the substring within the timeout period.
+   */
+  async waitForUrlContaining({ url, timeout = this.timeout }) {
+    return await this.driver.wait(until.urlContains(url), timeout);
```

Comment:
This return type of `this.driver.wait()` is `Promise<WebElement>`, but the JSDoc for the function signature of `waitForUrlContaining` returns `Promise<void>`.

I think you need to:

```suggestion
  await this.driver.wait(until.urlContains(url), timeout);
```